### PR TITLE
Use MutationObserver to observe log for work instance_linker.js correctly

### PR DESCRIPTION
**Pull request by inohiro**
instance_linker.js that makes hyperlink from child job instance to parent job instance had not worked correctly. This PR changes it works fine.

logs are appended to `#logs` element by jQuery. So we need to observe changes of the element with [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver).

I'm not sure about JavaScript, if someone have better idea to solve it, please give me comments.

### before

screen shot 2017-06-26 at 11 56 10

### after

screen shot 2017-06-26 at 13 02 45

### CODE REVIEW

**Comment by inohiro** [comment]
oh CI failed... I'll confirm it

File: app/assets/javascripts/kuroko2/instance_linker.js [hunk 1]
```diff
@@ -1,10 +1,25 @@
...
-	return '<a href="/definitions/' + jobId + '/instances/' + instanceId+ '">' + match + '</a>';
-      })
-    );
+  var MutationObserver = window.MutationObserver || window.WebKitMutationObserver || window.MozMutationObserver;
+  if (!MutationObserver) {
+    return;
+  }
+  var logParent = document.querySelector('#logs tbody');
+  var observer = new MutationObserver(function(mutations) {
+    if (mutations.some(function(m) {
```
Comment (eisuke) [inline app/assets/javascripts/kuroko2/instance_linker.js]: How about separate as a named function?

Comment (inohiro) [inline app/assets/javascripts/kuroko2/instance_linker.js]: I should do that 👍

File: app/assets/javascripts/kuroko2/instance_linker.js [hunk 2]
```diff
@@ -1,10 +1,25 @@
...
-  $('td.log').each(function(){
-    var logText = $(this).html();
-    $(this).html(
-      logText.replace(/instance#(\d+)\/(\d+)(?:\s+|\.)/, function(match, jobId, instanceId){
-	return '<a href="/definitions/' + jobId + '/instances/' + instanceId+ '">' + match + '</a>';
-      })
-    );
+  var MutationObserver = window.MutationObserver || window.WebKitMutationObserver || window.MozMutationObserver;
+  if (!MutationObserver) {
+    return;
```
Comment (inohiro) [inline app/assets/javascripts/kuroko2/instance_linker.js]: PhantomJS supports MutationObserver from 2.0. But It seems that TravisCI uses PhantomJS 1.x currently (as default). So if browser does not support MutationObserver, this function (instance linker) should be disabled.

File: app/assets/javascripts/kuroko2/instance_linker.js [hunk 3]
```diff
@@ -1,10 +1,25 @@
...
+    if (mutations.some(function(m) {
+      return m.addedNodes
+        && m.addedNodes instanceof NodeList
+        && m.addedNodes.length > 0
+        && m.type == 'childList'
+    })) {
+      $('td.log').each(function() {
+        var logText = $(this).html();
+        $(this).html(
+          logText.replace(/instance#(\d+)\/(\d+)(?:\s+|\.|$)/, function(match, jobId, instanceId) {
```
Comment (inohiro) [inline app/assets/javascripts/kuroko2/instance_linker.js]: FYI I added `|$`

File: app/assets/javascripts/kuroko2/instance_linker.js (hunk 2 above)
Comment (inohiro) [inline app/assets/javascripts/kuroko2/instance_linker.js]: But recent modern browsers support MutationObserver like that https://caniuse.com/#feat=mutationobserver

**Comment by eisuke** [comment]
I fix CI failing in https://github.com/cookpad/kuroko2/pull/69 🙏

**Comment by eisuke** [comment]
Merged #69. Please rebase this branch.

File: app/assets/javascripts/kuroko2/instance_linker.js (hunk 2 above)
Comment (eisuke) [inline app/assets/javascripts/kuroko2/instance_linker.js]: Thank you for your information.
To support PhantomJS 2.0 in https://github.com/cookpad/kuroko2/pull/70 .
Could you add a test after merged #70 ?

Comment (inohiro) [inline app/assets/javascripts/kuroko2/instance_linker.js]: Thank you about #70. I'll add a test about this.

**Comment by riseshia** [comment]
It seems ready to go. ping @eisuke ?

**Comment by eisuke** [comment]
👍

**Pull request merged**